zcncore/get_data.go

```diff
@@ -323,3 +368,102 @@ func GetUserLockedTotal(clientID string) (int64, error) {
 		return 0, err
 	}
 }
+
+// GetStakePoolUserInfo get stake pool user info for all blobbers/miners/sharders.
+// # Inputs
+//   - clientID wallet id
+func GetStakePoolUserInfo(clientID string) ([]byte, error) {
+	err := CheckConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	limit, offset := 20, 0
+	spUserInfo, err := sdk.GetStakePoolUserInfo(clientID, offset, limit)
+	if err != nil {
+		return nil, err
+	}
+
+	var spUserInfoResponse []*sdk.StakePoolDelegatePoolInfo
+
+	var spUserInfoSl []*sdk.StakePoolUserInfo
+	spUserInfoSl = append(spUserInfoSl, spUserInfo)
+	for {
+		// if the length of the slice is less than the limit, then we have reached the end
+		if len(spUserInfoSl) < limit {
+			break
+		}
+
+		// get the next set of stake pool user info
+		offset += limit
+		spUserInfo, err = sdk.GetStakePoolUserInfo(clientID, limit, offset)
+		if err != nil {
+			break
+		}
+		spUserInfoSl = append(spUserInfoSl, spUserInfo)
+	}
+
+	res, err := GetMinerSCUserInfo(clientID)
+	if err != nil {
+		return nil, errors.New("error while getting miner smart contract user info: " + err.Error())
+	}
+
+	var minerSCUserInfo *sdk.StakePoolUserInfo
+	err = json.Unmarshal(res, &minerSCUserInfo)
+	if err != nil {
+		return nil, errors.New("error while unmarshalling miner smart contract user info: " + err.Error())
+	}
+
+	spUserInfoSl = append(spUserInfoSl, minerSCUserInfo)
+
+	for _, pool := range spUserInfoSl {
+		for _, sp := range pool.Pools {
+			spUserInfoResponse = append(spUserInfoResponse, sp...)
+		}
+	}
+	response := map[string]interface{}{
+		"pools": spUserInfoResponse,
+	}
+
+	spUserInfoBytes, err := json.Marshal(response)
+	if err != nil {
+		return nil, errors.New("error while marshalling stake pool user info: " + err.Error())
+	}
+
+	return spUserInfoBytes, nil
+}
+
+func GetTransactions(toClientId, fromClientId, order string, limit, offset int64) ([]byte, error) {
+	if err := CheckConfig(); err != nil {
+		return nil, err
+	}
+
+	const GET_TRANSACTIONS = `/transactions`
+
+	return client.MakeSCRestAPICall(StorageSmartContractAddress, GET_TRANSACTIONS, Params{
+		"to_client_id": toClientId,
+		"client_id":    fromClientId,
+		"order":        order,
+		"limit":        strconv.FormatInt(limit, 10),
+		"offset":       strconv.FormatInt(offset, 10),
+	})
+}
+
+func GetFeesTable(reqPercentage float32) ([]byte, error) {
```

Comment:
Please make sure it is not duplicated. It should be implemented already.

Reply:
yes it's in transactions but arguments are different in  `transaction.GetFeesTable` I've to pass the miners as well which would not be ideal in case of mobile apps so in this method we only need to pass the requiredPercentage of the n/w to calculate the fees.

Reply:
Don't recreate it independently. You can build on the exisiting function so that updates to them are made at the same time.

Reply:
Same for this internally I'm using the GetFeesTable from core transactions.